extensions/lifecycle/management.js

```diff
@@ -68,14 +78,23 @@ function putLifecycleConfiguration(bucketName, workflows, cb) {
             }),
         },
     };
-    getS3Client(endpoint).putBucketLifecycleConfiguration(params, err => {
-        logger.debug('lifecycle configuration apply done', {
-            bucket: bucketName, error: err });
-        if (err && err.code === 'NoSuchBucket') {
+    
+    (async () => {
```

Comment:
Why you do that ? You can define the function as async directly ? And then remove your anonymous function ?